realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -86,116 +86,164 @@ void RealSenseNodeFactory::notification_handler(const rs2::notification &n, int
-                    }
-
-                    ROS_ERROR_STREAM("notification: Device " << _usb_port_id << " found... and was reset");
-                    found = true;
-                    break;
-                }
-            }
-            if(!found)
-            {
-                ROS_ERROR_STREAM("notification: Device " << _usb_port_id << " not found... and not reset");
-            }
+            setUpResinChuck();
         }
+        else
+        {
+            setUpNonResinChuck();
+        }
+    }
+    catch (const std::exception& ex)
+    {
+        ROS_ERROR_STREAM("An exception has been thrown: " << ex.what());
+        resetAndShutdown();
+    }
+    catch (...)
+    {
+        ROS_ERROR_STREAM("Unknown exception has occurred!");
+        resetAndShutdown();
+    }
+}
 
-        ROS_ERROR_STREAM("notification: Device " << _usb_port_id << " sleeping for 10 seconds");
-        boost::this_thread::sleep(boost::posix_time::seconds(10));
+void RealSenseNodeFactory::setUpResinChuck()
+{
+    auto privateNh = getPrivateNodeHandle();
+
+    privateNh.param("usb_port_id", _usb_port_id, std::string(""));
 
-        found = false;
+    {
+        std::lock_guard<std::recursive_mutex> scopedLock(_device_lock);
+
+        // Initial population of the device list
+        bool found = false;
         while (!found)
         {
             for (auto &&dev : _context.query_devices())
             {
                 if (deviceMatches(dev, _usb_port_id))
                 {
-                    try
-                    {
-                        addDevice(dev);
-                    }
-                    catch (...)
-                    {
-                        ROS_ERROR_STREAM("notification: Device " << _usb_port_id << " Unknown exception has occurred while calling addDevice on new device. exiting...");
-                        exit(1);
-                    }
-                    ROS_ERROR_STREAM("notification: Device " << _usb_port_id << " found... and was added");
+                    addDevice(dev);
                     found = true;
+                    ROS_ERROR_STREAM("notification: Device " << _usb_port_id << " found... and was added");
                     break;
                 }
             }
             if (!found)
             {
-                ROS_ERROR_STREAM("notification: Device " << _usb_port_id << " not found... sleeping for 2 seconds");
+                ROS_ERROR_STREAM("notification: No devices found for adding... " << _usb_port_id << " sleeping for 2 seconds and polling again");
                 boost::this_thread::sleep(boost::posix_time::seconds(2));
+                _context = rs2::context{};
             }
         }
-    }).detach();
-}
 
-void RealSenseNodeFactory::onInit()
+    }
+}
+void RealSenseNodeFactory::setUpNonResinChuck()
 {
-    try
+    auto privateNh = getPrivateNodeHandle();
```

Comment:
Would rather copy the older code if we can, otherwise we have to prove the implementation for the legacy non resin.io device is byte for byte identical or test at least 30x with the same result as the older non resin bots (same %).